content/en/docs/concepts/services-networking/service-traffic-policy.md

```diff
@@ -0,0 +1,66 @@
+---
+reviewers:
+- maplain
+title: Service Internal Traffic Policy
+content_type: concept
+---
+
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+_Service Internal Traffic Policy_ enables internal traffic restrictions to only route
+internal traffic to endpoints within the node the traffic originated from. The
+"internal" traffic here refers to traffic originated from Pods in the current
+cluster. This can help to reduce costs and improve performance.
+
+<!-- body -->
+
+## Using Service Internal Traffic Policy
+
+You can enable Internal Traffic Policy for a Service by setting the
+`spec.internalTrafficPolicy` to "Local". This tells kube-proxy to only use node
+local endpoints for cluster internal traffic. Importantly, this Service will
+appear to have zero endpoint for any Pod accessing this Service on a node, when
+this Service doesn't have any local endpoint on the node, but does have endpoints
+on other nodes.
+
+The following example shows what a Service looks like when internalTrafficPolicy
+is set to "Local":
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  internalTrafficPolicy: Local
+```
+
+## How it Works
+
+kube-proxy filters the endpoints it routes to based on the
+`spec.internalTrafficPolicy` setting. When it's "Local", only node local
+endpoints are considered. When it's "Cluster" or missing, all endpoints are
+considered.
+When the feature gate `ServiceInternalTrafficPolicy` is on,
+`spec.internalTrafficPolicy` defaults to "Cluster".
+
+## Constraints
+
+* Service Internal Traffic Policy are not used when `externalTrafficPolicy` is set
+  to "Local" on a Service. It is possible to use both features in the same cluster
```

Comment:
I don't recall this corner-case.  It feels wrong - we're going to fix this, right?

Reply:
this is from https://github.com/kubernetes/website/pull/27032#discussion_r595420521 and https://github.com/kubernetes/kubernetes/issues/100313

Reply:
ACK thanks :)